app/main.py

```diff
@@ -2,0 +1,21 @@
+from typing import Union
+
+from app.cafe import Cafe
+from app.errors import VaccineError, NotWearingMaskError
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> Union[str, None]:
+    masks_needed = 0
+
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            masks_needed += 1
+
+    if masks_needed:
+        return f"Friends should buy {masks_needed} masks"
```

Comment:
Code Efficiency: There is no need to compare `masks_needed` to `0` here. In Python, `0` is considered `False` and any other number is considered `True` in a boolean context. So you can simply write `if masks_needed:`. It will make your code cleaner.